src/crypto/ShortHash.cpp

```diff
@@ -3,31 +3,66 @@
 // of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
 
 #include "ShortHash.h"
+#include "fmt/format.h"
+#include <mutex>
 #include <sodium.h>
 
 namespace stellar
 {
 namespace shortHash
 {
-static unsigned char sKey[crypto_shorthash_KEYBYTES];
+static unsigned char gKey[crypto_shorthash_KEYBYTES];
+static std::mutex gKeyMutex;
```

Comment:
I'm not 100% sure how I feel about this. We were previously treating this as initialization is not thread safe, and therefore there wasn't any locking overhead during normal operation. I don't have a good feeling for how big the impact of this will be. I don't know what the alternatives are. Thoughts?

Reply:
I just tested this: cranked the `[shorthash]` test iteration count up to 100000. Even then, `perf(1)` isn't capable of measuring non-noise differences in time (it actually seems to think the mutex version goes _faster_, but it's well within noise) and the instruction count difference is 11711107277 insns without mutex, 11712128476 with mutex (i.e. 0.00008x more instructions executed, on a test that does nothing but call shorthash in a loop).

I guess non-contended mutex acquire is fairly fast these days!